src/lib/loaders/loaders_with_authentication/vortex.ts

```diff
@@ -34,15 +36,98 @@ export default (accessToken, opts) => {
     { method: "POST" }
   )
 
+  const vortexGraphqlLoader = ({ query, variables }: LoaderArgs) =>
+    vortexLoader(
+      "/graphql",
+      { query, variables: JSON.stringify(variables) },
+      {
+        method: "POST",
+      }
+    )
+
   return {
     vortexTokenLoader,
-    vortexGraphqlLoader: ({ query, variables }: LoaderArgs) =>
-      vortexLoader(
-        "/graphql",
-        { query, variables: JSON.stringify(variables) },
-        {
-          method: "POST",
+    vortexGraphqlLoader,
+    marketPriceInsightsBatchLoader: async (
+      params: { artistId: string; medium: string; category: string }[]
+    ) => {
+      // TODO: Fix this logic once we only need category to fetch insights
+      const artistIDMediumTuples = params.map((artist) => ({
+        artistId: artist.artistId,
+        medium: getVortexMedium(artist.medium, artist.category),
+      }))
+
+      try {
+        const vortexResult = await vortexGraphqlLoader({
+          query: gql`
+        query MarketPriceInsightsBatchQuery($artistIDMediumTuples: [ArtistIdMediumTupleType!]!) {
+          marketPriceInsightsBatch(input: $artistIDMediumTuples, first: ${MAX_MARKET_PRICE_INSIGHTS}) {
+            totalCount
+            edges {
+              node {
+                artistId
+                medium
+                demandRank
+                annualLotsSold
+                annualValueSoldCents
+                lastAuctionResultDate
+              }
+            }
+          }
         }
-      ),
+      `,
+          variables: { artistIDMediumTuples },
+        })()
+
+        const priceInsightNodes: MarketPriceInsight[] = extractNodes(
+          vortexResult.data?.marketPriceInsightsBatch
+        )
+
+        return priceInsightNodes
+      } catch (e) {
+        console.error(e)
```

Comment:
Log to Sentry?

Reply:
Yes! I've moved the try-catch block outside of the loader to the function [enrichArtworksWithPriceInsights](https://github.com/artsy/metaphysics/blob/b7de1a3118724ec0b480f703b880e9c297b8be59/src/schema/v2/me/myCollection.ts#L272) and added a `Sentry.captureException`.